### PR TITLE
Use topic params when displaying new form.

### PR DESCRIPTION
**Pull request by timdiggins**
This allows linkages to new topic with preset content/title
implements https://github.com/thredded/thredded/issues/297

### CODE REVIEW

File: app/controllers/thredded/topics_controller.rb [hunk 1]
```diff
@@ -109,7 +109,10 @@ def topic_params
         .merge(user: thredded_current_user)
     end
 
-    def new_topic_params
+    def new_topic_params(allow_empty: false)
+      if allow_empty && params[:topic].blank?
+        return { messageboard: messageboard, user: thredded_current_user }
+      end
       params
         .require(:topic)
```
Comment (glebm) [inline app/controllers/thredded/topics_controller.rb]: Let's just replace this `require(:topic)` with `fetch(:topic, {})`, making the `allow_empty` param unnecessary.

**Comment by timdiggins** [comment]
@glebm agreed, done

**Comment by glebm** [comment]
Looks good! Please squash the commits and I'll merge :+1:

**Comment by timdiggins** [comment]
@glebm done (though -- do you know about the recent github squash and merge PR? )

**Comment by glebm** [comment]
@timdiggins Thanks! We haven't enabled the squash & merge feature yet, but will try it out. Looks like it delegates the responsibility of writing the squashed commit message to the repo maintainers which is not ideal.

**Pull request merged**